src/classes/Trades.ts

```diff
@@ -639,6 +694,431 @@ export default class Trades {
         });
     }
 
+    private counterOffer(offer: TradeOffer, meta: Meta): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const start = dayjs().valueOf();
+
+            const opt = this.bot.options;
+
+            const theirInventory = new Inventory(
+                offer.partner,
+                this.bot.manager,
+                this.bot.schema,
+                opt,
+                this.bot.effects,
+                this.bot.paints,
+                this.bot.strangeParts,
+                'their'
+            );
+
+            const ourInventoryItems = this.bot.inventoryManager.getInventory.getItems;
+
+            log.debug('Fetching their inventory...');
+            void theirInventory.fetch().asCallback(err => {
+                if (err) {
+                    return reject(
+                        new Error(
+                            `Failed to load inventories (Steam might be down, or private inventory): ${JSON.stringify(
+                                err
+                            )}`
+                        )
+                    );
+                }
+
+                const ourItems = Inventory.fromItems(
+                    this.bot.client.steamID || this.bot.community.steamID,
+                    offer.itemsToGive,
+                    this.bot.manager,
+                    this.bot.schema,
+                    opt,
+                    this.bot.effects,
+                    this.bot.paints,
+                    this.bot.strangeParts,
+                    'our'
+                ).getItems;
+
+                const theirItems = Inventory.fromItems(
+                    offer.partner,
+                    offer.itemsToReceive,
+                    this.bot.manager,
+                    this.bot.schema,
+                    opt,
+                    this.bot.effects,
+                    this.bot.paints,
+                    this.bot.strangeParts,
+                    'their'
+                ).getItems;
+
+                const theirInventoryItems = theirInventory.getItems;
+
+                log.debug('Set counteroffer...');
+                const counter = offer.counter();
+
+                const showOnlyMetal = opt.miscSettings.showOnlyMetal.enable;
+                // To the person who thinks about changing it. I have a gun keep out ( う-´)づ︻╦̵̵̿╤── \(˚☐˚”)/
+                // Extensive tutorial if you want to update this function https://www.youtube.com/watch?v=dQw4w9WgXcQ.
+
+                log.debug('Set counteroffer message...');
+                counter.setMessage(
+                    "Oni-chan. I'm not a dummie (thicc) offer contains wrong value. You've probably made a few mistakes, here's the correct offer."
```

Comment:
I will not approve until this is changed or implement an option for a custom message 😚

Reply:
![Screenshot_131](https://user-images.githubusercontent.com/11471265/119862583-5244d200-bf21-11eb-846c-999070388c19.png)
![654](https://user-images.githubusercontent.com/11471265/119862183-e95d5a00-bf20-11eb-8475-348ed267090d.jpg)

Reply:
🤣🤣🤣